keps/487-kubectl-plugin/README.md

```diff
@@ -0,0 +1,573 @@
+
+The above should also provide different output formats:
+
+```bash
+kuctl describe taco-123 -o json
+kuctl describe taco-123 -o yaml
+```
+
+### Cancel Workloads
+
+A request to cancel would be akin to deleting the CRD. A "cancel" is more intuitive / natural than a delete request for this use case.
+This implementation will be tricky because we need to make the request to the underlying controller.
+
+
+```bash
+kuctl cancel taco-123
+```
+
+It might also be useful to request a cancel all, limited to the permission that the user has, a namespace, or other filter. This likely needs a confirmation.
+
+```bash
+kuctl cancel --all
+> Are you sure you want to cancel all workloads y/n?
+```
+
+Or without the prompt:
+
+```bash
+kuctl cancel --all --force
+```
+
+Or within a specific filter:
+
+```bash
+kuctl cancel --all --namespace insects
+```
+
+To support the multi-tenancy use case, filters for each of local-queue, cluster-queue and cohort will be allowed.
+
+### Watch Workload
+
+After submitting a workload, it's nice to be able to watch / stream logs. That should be easy to do.
+
+```bash
+kuctl watch taco-123
+```
+```console
+... makin' tacos!
+... taco 1 is cooked!
+... taco 2 is cooking.
+```
+
+Or a user may want to watch or stream events:
+
+kuctl watch taco-123 --events
+```
+```console
+<timestamp> <event1>
+<timestamp> <event2>
+...
+```
+
+
+### Queues
+
+I haven't used queues extensively so this likely needs to be expanded, but akin to listing workloads, we probably want to list kueues. For all queues, this could be:
+
+```bash
+kuctl queues
+```
+
+| Name | Admitted | Pending | State |
+|------|-----------|---------|------|
+| user-queue | 1 | 1 | Operational |
+
+To filter to a specific queue:
+
+```bash
+kuctl queues user-queue
+```
+
+We likely want to be able to provide more metadata, and either we could add a `describe-queue` subcommand, or have the above second example show the more verbose information. I think I like the first idea. We also likely want to expand these subcommands to include details that the original prompt warranted. I'm not familiar enough with Kueue yet to add them. This command should also support yaml/json.
+
+```bash
+kuctl queues -o json
+```
+
+
+### Future Commands
+
+In the future, if we can make this a full fledged client for interaction with workloads, we could consider the following commands.
+Note that these are not proposed to be in the first stage of this design document.
+
+#### Submit
+
+```
+# Submit, either a yaml as is
+kuctl submit workload.yaml
+
+# or a simpler abstraction that uses some kind of default or template
+# This would actually be really cool if we could map a community develoepd job or workload spec (that works for other tools) into kueue
+kuctl submit <something else>
+```
+
+#### Update
```

Comment:
When I've used edit, it typically opens the yaml in an editor. Update would be a more scoped set of commands to update key value pairs without the editor. It would be more akin to how helm install does set with keys and values.

Reply:
oh, it was not `edit`, it was `patch` https://jamesdefabia.github.io/docs/user-guide/kubectl/kubectl_patch/